compiler/src/dotty/tools/dotc/core/OrderingConstraint.scala

```diff
@@ -248,7 +248,7 @@ class OrderingConstraint(private val boundsMap: ParamBounds,
       NoType
     case tp: TypeBounds =>
       val lo1 = stripParams(tp.lo, todos, !isUpper).orElse(defn.NothingType)
-      val hi1 = stripParams(tp.hi, todos, isUpper).orElse(defn.AnyKindType)
+      val hi1 = stripParams(tp.hi, todos, isUpper).orElse(tp.kind)
```

Comment:
You should be able to use [`TypeApplications#topType`](https://github.com/lampepfl/dotty/blob/fa3893e96ea8e1d3beaedcc17bec67ce74ca799f/compiler/src/dotty/tools/dotc/core/TypeApplications.scala#L232-L240) instead which I needed for the same reason :).
```suggestion
      val hi1 = stripParams(tp.hi, todos, isUpper).orElse(tp.topType)
```